### ISSUE TITLE
Cannot execute x86 container on x86_64 host

### ISSUE THREAD

**Issue by tenforward** [lxc/lxc#2208]
# Required information

 * Distribution: Plamo
 * Distribution version: 7
 * The output of
   * `lxc-start --version` 3.0.0.beta1
   * `lxc-checkconfig`

```
--- Namespaces ---
Namespaces: enabled
Utsname namespace: enabled
Ipc namespace: enabled
Pid namespace: enabled
User namespace: enabled
Network namespace: enabled

--- Control groups ---
Cgroups: enabled

Cgroup v1 mount points: 
/sys/fs/cgroup/cpuset
/sys/fs/cgroup/cpu
/sys/fs/cgroup/cpuacct
/sys/fs/cgroup/blkio
/sys/fs/cgroup/memory
/sys/fs/cgroup/devices
/sys/fs/cgroup/freezer
/sys/fs/cgroup/net_cls
/sys/fs/cgroup/perf_event
/sys/fs/cgroup/net_prio
/sys/fs/cgroup/hugetlb
/sys/fs/cgroup/pids
/sys/fs/cgroup/rdma
/sys/fs/cgroup/debug

Cgroup v2 mount points: 


Cgroup v1 systemd controller: missing
Cgroup v1 clone_children flag: enabled
Cgroup device: enabled
Cgroup sched: enabled
Cgroup cpu account: enabled
Cgroup memory controller: enabled
Cgroup cpuset: enabled

--- Misc ---
Veth pair device: enabled, loaded
Macvlan: enabled, not loaded
Vlan: enabled, not loaded
Bridges: enabled, loaded
Advanced netfilter: enabled, not loaded
CONFIG_NF_NAT_IPV4: enabled, loaded
CONFIG_NF_NAT_IPV6: enabled, not loaded
CONFIG_IP_NF_TARGET_MASQUERADE: enabled, loaded
CONFIG_IP6_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_NETFILTER_XT_TARGET_CHECKSUM: enabled, loaded
CONFIG_NETFILTER_XT_MATCH_COMMENT: enabled, not loaded
FUSE (for use with lxcfs): enabled, not loaded

--- Checkpoint/Restore ---
checkpoint restore: enabled
CONFIG_FHANDLE: enabled
CONFIG_EVENTFD: enabled
CONFIG_EPOLL: enabled
CONFIG_UNIX_DIAG: enabled
CONFIG_INET_DIAG: enabled
CONFIG_PACKET_DIAG: enabled
CONFIG_NETLINK_DIAG: enabled
File capabilities: 

Note : Before booting a new kernel, you can check its configuration
usage : CONFIG=/path/to/config /usr/bin/lxc-checkconfig
```

   * `uname -a` 

`Linux enterprise 4.15.3-plamo64 #3 SMP PREEMPT Thu Mar 8 22:48:38 JST 2018 x86_64 GNU/Linux`
   * `cat /proc/self/cgroup`
```
14:debug:/
13:rdma:/
12:pids:/
11:hugetlb:/
10:net_prio:/
9:perf_event:/
8:net_cls:/
7:freezer:/
6:devices:/
5:memory:/
4:blkio:/
3:cpuacct:/
2:cpu:/
1:cpuset:/
```
   * `cat /proc/1/mounts`
```
/dev/root / ext4 rw,relatime,data=ordered 0 0
devtmpfs /dev devtmpfs rw,relatime,size=3981604k,nr_inodes=995401,mode=755 0 0
tmpfs /run tmpfs rw,relatime 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
/dev/sdb4 /home ext4 rw,relatime,data=ordered 0 0
/dev/mapper/kernvg-kern01 /usr/src btrfs rw,noatime,compress=lzo,ssd,space_cache,subvolid=5,subvol=/ 0 0
/dev/mapper/lxcvg-lxclv01 /var/lib/lxc btrfs rw,noatime,compress=lzo,ssd,space_cache,subvolid=257,subvol=/lxc 0 0
devpts /dev/pts devpts rw,relatime,gid=5,mode=620,ptmxmode=000 0 0
/etc/auto.net /net autofs rw,relatime,fd=6,pgrp=4416,timeout=300,minproto=5,maxproto=5,indirect,pipe_ino=6489 0 0
voyager:/backup /backup nfs4 rw,relatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=172.16.43.183,local_lock=none,addr=172.16.43.182 0 0
cgroup /sys/fs/cgroup tmpfs rw,relatime,mode=755 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/cpu cgroup rw,relatime,cpu 0 0
cgroup /sys/fs/cgroup/cpuacct cgroup rw,relatime,cpuacct 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,relatime,blkio 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,relatime,memory 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,relatime,devices 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,relatime,freezer 0 0
cgroup /sys/fs/cgroup/net_cls cgroup rw,relatime,net_cls 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/net_prio cgroup rw,relatime,net_prio 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,relatime,hugetlb 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,relatime,pids 0 0
cgroup /sys/fs/cgroup/rdma cgroup rw,relatime,rdma 0 0
cgroup /sys/fs/cgroup/debug cgroup rw,relatime,debug 0 0
none /run/user/1000 tmpfs rw,relatime,mode=700,uid=1000 0 0
none /run/user/0 tmpfs rw,relatime,mode=700 0 0
```

# Issue description

I can't run a x86 container with `lxc.arch = x86` on x86_64 host.

I compile lxc-3.0.0.beta1 from source on Plamo 7.0(-dev), and create a x86 container.

I start the container, and execute `uname -m` in the container. I expect to get i686, but I get x86_64.

```
# lxc-create -n alpine -t download -- -d alpine -r 3.4 -a i386
# grep lxc.arch /var/lib/lxc/alpine/config 
lxc.arch = x86
# lxc-start -n alpine -o log -l debug
# lxc-attach -n alpine -- uname -m
x86_64
```

The container log has a line like the following:
```
      lxc-start alpine 20180309064630.304 WARN     lxc_confile - /home/karma/Plamo-src/contrib/container/lxc/lxc-3.0.0.beta1/src/lxc/confile.c:set_config_personality:969 - Unsupported personality "x86"
```

I assume that `HAVE_SYS_PERSONALITY_H` is not defined in src/lxc/tools/tools_utils.c. So I try applying the patch, as follows:

```
diff -uNr lxc-3.0.0.beta1.orig/src/lxc/tools/tool_utils.c lxc-3.0.0.beta1/src/lxc/tools/tool_utils.c
--- lxc-3.0.0.beta1.orig/src/lxc/tools/tool_utils.c	2018-03-02 03:59:49.000000000 +0900
+++ lxc-3.0.0.beta1/src/lxc/tools/tool_utils.c	2018-03-09 16:33:02.859204300 +0900
@@ -37,6 +37,8 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 
+#include "config.h"
+
 #if HAVE_SYS_PERSONALITY_H
 #include <sys/personality.h>
 #endif
```

and recompile lxc, and execute a container:

```
# lxc-start -n alpine -o log -l debug
# lxc-attach -n alpine -- uname -m
i686
```

# Other Information

For confirmation, I installed 3.0.0.beta1 package from PPA to Ubuntu xenial. And I execute the container in the same way. I get "i686".

```
# dpkg -l | grep lxc
ii  liblxc-common                       3.0.0~beta1+master~20180301-2211-0ubuntu1~xenial amd64        Linux Containers userspace tools (common tools)
ii  liblxc1                             3.0.0~beta1+master~20180301-2211-0ubuntu1~xenial amd64        Linux Containers userspace tools (library)
ii  lxc                                 3.0.0~beta1+master~20180301-2211-0ubuntu1~xenial all          Transitional package - lxc -> lxc-utils
ii  lxc-templates                       3.0.0~beta1-0ubuntu1~ubuntu16.04.1~ppa1          amd64        Linux Containers userspace tools (templates)
ii  lxc-utils                           3.0.0~beta1+master~20180301-2211-0ubuntu1~xenial amd64        Linux Containers userspace tools
ii  lxcfs                               3.0.0~beta1-0ubuntu1~ubuntu16.04.1~ppa1          amd64        FUSE based filesystem for LXC
# lxc-create -t download -n alpine -- -d alpine -r 3.4 -a i386
# lxc-start -n alpine -o log -l debug
# lxc-attach -n alpine -- uname -m
i686
```

Is this a bug? or is there something wrong with my compile environment?

**Comment by brauner** [comment]
@tenforward can you send a patch?

**Comment by tenforward** [comment]
Yes!

**Comment by tenforward** [comment]
I wonder why the container could be executed correctly the PPA package on Ubuntu. (?_?) :sweat_smile:

**Comment by brauner** [comment]
Closed by #2209.
